tray/monitor.go

```diff
@@ -258,12 +262,15 @@ func (ti *Instance) pollingMonitor() {
 			}
 		}
 
-		if initialChan != nil {
-			initialChan <- struct{}{}
-			close(initialChan)
-			initialChan = nil
-			if ti.debugMode {
-				log.Println(internal.DebugPrefix, "Initial retrieve")
+		// while the settings were not fetch don't unblock the tray loop
+		if ti.state.trayStatus != Invalid {
+			if initialChan != nil {
```

Comment:
Can be combined to a single `if initialChan != nil && ti.state.trayStatus != Invalid`